test/functional/test_framework/test_node.py

```diff
@@ -542,6 +542,30 @@ def add_p2p_connection(self, p2p_conn, *, wait_for_verack=True, **kwargs):
 
         return p2p_conn
 
+    def add_outbound_p2p_connection(self, p2p_conn, *, connection_type="outbound", connect_id=0, **kwargs):
+        """ Add an outbound p2p connection from node. Either
+        full-relay("outbound") or block-relay-only("blockrelay") connection.
+
+        This method adds the p2p connection to the self.p2ps list and also
+        returns the connection to the caller."""
+
+        kwargs['dstport'] = p2p_port(self.index)
```

Comment:
good point. I actually ended up removing `dstport` and `dstaddr` from the `peer_accept_connection` function signature, because they aren't being set here (we don't know the exact info until the listening server is spun up, which is why we pass a callback through)- so it seems more like a red herring.